src/utils.jl

```diff
@@ -430,19 +433,19 @@ julia> map(splitarg, (:(f(a=2, x::Int=nothing, y, args...))).args[2:end])
 See also: [`combinearg`](@ref)
 """
 function splitarg(arg_expr)
-    splitvar(arg) =
-        (@match arg begin
-            ::T_ => (nothing, T)
-            name_::T_ => (name, T)
-            x_ => (x, :Any)
-        end)::NTuple{2,Any} # the pattern `x_` matches any expression
-    (is_splat = @capture(arg_expr, arg_expr2_...)) || (arg_expr2 = arg_expr)
-    if @capture(arg_expr2, arg_ = default_)
-        @assert default !== nothing "splitarg cannot handle `nothing` as a default. Use a quoted `nothing` if possible. (MacroTools#35)"
-        return (splitvar(arg)..., is_splat, default)
+    if @capture(arg_expr, arg_expr2_ = default_)
+      @assert default !== nothing "splitarg cannot handle `nothing` as a default. Use a quoted `nothing` if possible. (MacroTools#35)"
```

Comment:
```suggestion
      # This assert will only be triggered if a `nothing` literal was somehow spliced into the Expr.
      # A regular `nothing` default is a `Symbol` object when it gets here. See #178 
      @assert default !== nothing "splitarg cannot handle `nothing` as a default. Use a quoted `nothing` if possible. (MacroTools#35)"
```

Reply:
It always trips me up that in Julia, `true` is a literal, but `nothing` isn't...
```julia
julia> dump(:((x, nothing, true)))
Expr
  head: Symbol tuple
  args: Array{Any}((3,))
    1: Symbol x
    2: Symbol nothing
    3: Bool true
```